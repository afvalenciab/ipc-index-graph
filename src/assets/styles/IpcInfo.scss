@import url('https://fonts.googleapis.com/css?family=Nunito+Sans:300,400,600,700,900&display=swap');

body {
  margin: 0;
  font-family: 'Nunito Sans', Arial;
  background-color: #F4F6F9;
}

.container--ipc {
  display: grid;
  grid-template: 80px 100px minmax(auto, 1fr) 30px / 100vw;
  grid-gap: 10px;
  min-height: 100vh;
  box-sizing: border-box;
}