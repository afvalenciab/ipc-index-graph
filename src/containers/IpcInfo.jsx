import React from 'react';
import { connect } from 'react-redux';
import Header from '../components/Header';
import IndexInfo from '../components/IndexInfo';
import IndexGraph from '../components/IndexGraph';
import InitMessage from '../components/InitMessage';
import Footer from '../components/Footer';
import '../assets/styles/IpcInfo.scss';

const IpcInfo = (props) => {
  const { historicalIpcList, user } = props;
  const hasUser = Object.keys(user).length > 0;

  return (
    <main className='container--ipc'>
      <Header />
      {hasUser ? (
        historicalIpcList.length > 0 && <IndexInfo />
      ) : (
        <hr style={{ border: '0' }} />
      )}

      {hasUser ? (
        historicalIpcList.length > 0 ? <IndexGraph /> : 'Loading...'
      ) : (
        <InitMessage />
      )}
      <Footer />
    </main>
  );
};

const mapStateToProps = (state) => {
  return {
    historicalIpcList: state.historicalIpcList,
    user: state.user,
  };
};

export default connect(mapStateToProps, null)(IpcInfo);
