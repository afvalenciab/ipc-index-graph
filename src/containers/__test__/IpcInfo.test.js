import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMocks';
import IpcInfo from '../IpcInfo';

describe('IpcInfo testing', () => {
  test('Match Snapshot', () => {
    const ipcInfo = create(
      <ProviderMock>
        <IpcInfo />
      </ProviderMock>
    );
    expect(ipcInfo.toJSON()).toMatchSnapshot();
  });
});
