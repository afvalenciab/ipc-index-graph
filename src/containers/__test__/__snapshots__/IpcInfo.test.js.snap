// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IpcInfo testing Match Snapshot 1`] = `
<main
  className="container--ipc"
>
  <header
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1>
        IPC
        <span>
          Index
        </span>
      </h1>
    </a>
    <nav
      className="menu"
    >
      <ul>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Ingresar
          </a>
        </li>
        <li
          className="menu--register"
        >
          <a
            href="/register"
            onClick={[Function]}
          >
            Crear cuenta
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <hr
    style={
      Object {
        "border": "0",
      }
    }
  />
  <section
    className="init__container"
  >
    <p
      className="init__description"
    >
      El Índice de Precios y Cotizaciones
      <span>
         (IPC) 
      </span>
      es el indicador oficial de la evolución del mercado accionario, ya que busca medir el rendimiento de las acciones de mayor tamaño y liquidez listadas en la Bolsa Mexicana de Valores.
    </p>
    <p
      className="init__invitation"
    >
      Ingresa y conoce el comportamiento del índice
    </p>
    <button
      onClick={[Function]}
      type="button"
    >
      Ingresar
    </button>
  </section>
  <footer
    className="footer"
  >
    <p>
      IPC Index - Indice de Precios y Cotizaciones.
    </p>
    <p>
      Hecho por Andres Valencia
    </p>
  </footer>
</main>
`;
